var/spack/repos/builtin/packages/yambo/package.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Yambo(AutotoolsPackage):
+    """Yambo is a FORTRAN/C code for Many-Body calculations in solid
+    state and molecular physics.
+
+    Yambo relies on the Kohn-Sham wavefunctions generated by two DFT
+    public codes: abinit, and PWscf. The code was originally developed
+    in the Condensed Matter Theoretical Group of the Physics Department
+    at the University of Rome "Tor Vergata" by Andrea Marini. Previous
+    to its release under the GPL license, yambo was known as SELF.
+    """
+
+    homepage = "http://www.yambo-code.org/index.php"
+    url = "https://github.com/yambo-code/yambo/archive/4.1.3.tar.gz"
+
+    version('4.2.1', '99027014192c0f0f4b5d9b48414ad85d')
+    version('4.2.0', '0cbb4d7c9790596d163ebe872d95bd30')
+
+    variant('dp', default=False, description='Enable double precision')
+    variant(
+        'profile',
+        values=('time', 'memory'),
+        default='',
+        description='Activate profiling of specific sections',
+        multi=True
+    )
+
+    variant(
+        'io',
+        values=('iotk', 'etsf-io'),
+        default='',
+        description='Activate support for different io formats (requires network access)',  # noqa
+        multi=True
+    )
+
+    # MPI + OpenMP parallelism
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('openmp', default=False, description='Enable OpenMP support')
+
+    depends_on('blas')
+    depends_on('lapack')
+
+    # MPI dependencies are forced, until we have proper forwarding of variants
+    depends_on('mpi', when='+mpi')
+    depends_on('netcdf+mpi', when='+mpi')
+    depends_on('hdf5+mpi', when='+mpi')
+    depends_on('fftw+mpi', when='+mpi')
+    depends_on('scalapack', when='+mpi')
+
+    depends_on('netcdf~mpi', when='~mpi')
```

Comment:
do you actually need to ask for `~mpi` here, it may work with just `netcdf`, right?

Reply:
The idea here is that if I say no MPI, then it should be no MPI all the way down :smile:

Reply:
yeah, i know, just saying it's not a requirement per-se. Just in case someone tries to use no-MPI `yamboo` with some other package which uses `netcdf+mpi`.

Reply:
Ok, I checked and I think I'll keep it like that. As far as I was told `yambo` is used as an application, and not linked as a library, thus there will be no case where another package pulls-in `netcdf+mpi` and wants to depend on `yambo~mpi`.

Reply:
ok, maybe add a super brief note above with this comment?